db/external_sst_file_test.cc

```diff
@@ -2063,6 +2094,64 @@ TEST_F(ExternalSSTFileTest, LinkExternalSst) {
   }
 }
 
+/*
+ * Test and verify the functionality of ingestion_options.move_files and
+ * ingestion_options.failed_move_fall_back_to_copy
+ */
+TEST_P(ExternSSTFileLinkFailFallbackTest, LinkFailFallBackExternalSst) {
+  Options options = CurrentOptions();
+  options.disable_auto_compactions = true;
+  const EnvOptions env_options;
+  DestroyAndReopen(options);
+  const int kNumKeys = 10000;
+  IngestExternalFileOptions ifo;
+  ifo.move_files = true;
+  ifo.failed_move_fall_back_to_copy = GetParam();
+
+  std::string file_path = sst_files_dir_ + "file1.sst";
+  // Create SstFileWriter for default column family
+  SstFileWriter sst_file_writer(env_options, options);
+  ASSERT_OK(sst_file_writer.Open(file_path));
+  for (int i = 0; i < kNumKeys; i++) {
+    ASSERT_OK(sst_file_writer.Put(Key(i), Key(i) + "_value"));
+  }
+  ASSERT_OK(sst_file_writer.Finish());
+  uint64_t file_size = 0;
+  ASSERT_OK(env_->GetFileSize(file_path, &file_size));
+
+  ExternalSSTTestEnv test_env(options.env, true);
```

Comment:
I think we need to set `options.env = new ExternalSSTTestEnv` before opening the db.

Reply:
I updated the code as you suggested.